### DISCUSSION TITLE
我把vant的源码下载到本地了，想着运行起来，但是总是报错

### DISCUSSION THREAD

**Discussion by dabaoRain** [youzan/vant#9838] [Q&A] [answered]
### What problem does this feature solve?

vant源码本地运行启动

### What does the proposed API look like?

TypeError: Cannot destructure property 'nav' of 'vantConfig.site' as it is undefined. 在控制台执行的时候报这样的错误，我更换了node版本，删除node_modules文件等操作后仍然运行不起来。各位都是如何运行启动的，麻烦指点下

**Comment by gunsley** [comment]
我也遇到这个问题了，求解

**Comment by chenjiahan** [comment]
可以参考下 https://vant-contrib.gitee.io/vant/v3/#/zh-CN/contribution#ben-di-kai-fa

**Comment by sonOfBigBrother** [comment, marked as answer]
我也遇到相同问题，下载的是 dev 分支，按照错误日志提示，找到了可能出错的脚本代码，加了一句输出错误日志代码。
```javascript
// packages/vant-cli/lib/common/constant.js

sync function getVantConfigAsync() {
    const require = createRequire(import.meta.url);
    delete require.cache[VANT_CONFIG_FILE];
    try {
        return (await import(VANT_CONFIG_FILE)).default;
    }
    catch (err) {
       // 自己加的
        console.error(err)
        return {};
    }
}
const vantConfig = await getVantConfigAsync();
export function getVantConfig() {
    return vantConfig;
}
```
运行`pnpm run dev`后在控制台看到了这个报错信息：
> Error [ERR_UNSUPPORTED_ESM_URL_SCHEME]: Only file and data URLs are supported by the default ESM loader. On Windows, absolute paths must be valid file:// URLs. Received protocol 'c:'

我的日常办公环境是 Windows 系统，绝对路径是不带 file 协议且以磁盘盘符开头；按照这个提示，我又在 Linux（CentOS 7.6）上测试——**项目运行是没问题的**。
查阅了下 node 官网，找到了 API `url.pathToFileURL`，添加到如下代码中
```javascript
return (await import(pathToFileURL(VANT_CONFIG_FILE))).default;
```
再次执行`pnpm run dev`后就可以正常起项目了。
这是我提供一个的解决方案，希望官方能核查这个问题是否在 Windows 上必现。

**Reply by chenjiahan** [reply]
感谢协助排查，这个确实是 dynamic import 在 windows 上的兼容问题，目前已经在 dev 分支修复了，可以重新 pull 代码试一下

**Reply by chenjiahan** [reply]
PR：https://github.com/youzan/vant/pull/9898
